ocaml/xapi/updateinfo.ml

```diff
@@ -558,6 +579,32 @@ module UpdateInfo = struct
                           }
                       | Xml.Element ("livepatches", _, livepatches) ->
                           {acc with livepatches= LivePatch.of_xml livepatches}
+                      | Xml.Element ("issued", attr, _) ->
+                          let issued =
+                            match List.assoc_opt "date" attr with
+                            | Some date -> (
+                              try
+                                Xapi_stdext_date.Date.of_string
+                                  (Scanf.sscanf date
+                                     "%04d-%02d-%02d %02d:%02d:%02d"
+                                     (fun y mon d h m s ->
+                                       Printf.sprintf
+                                         "%04i%02i%02iT%02i:%02i:%02iZ" y mon d
+                                         h m s
+                                   )
+                                  )
+                              with e ->
+                                (* The error should not block update. Ingore it
+                                   and set "issued" as epoch. *)
+                                warn "%s" (ExnHelper.string_of_exn e) ;
+                                Xapi_stdext_date.Date.epoch
+                            )
+                            | None ->
+                                Xapi_stdext_date.Date.epoch
+                          in
+                          {acc with issued}
+                      | Xml.Element ("severity", _, [Xml.PCData v]) ->
+                          {acc with severity= Severity.of_string v}
```

Comment:
This may raise exception as well. But this should not block the update. So I would suggest to add a warning log and ignore the error with a default value.

Reply:
Sorry, I missed that! Added the exception handler.